tests/LogNonGenuineHostTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace MongoDB\Tests;
+
+use MongoDB\Client;
+use MongoDB\Driver\Exception\InvalidArgumentException;
+use Psr\Log\AbstractLogger;
+use Psr\Log\LoggerInterface;
+
+use function func_get_args;
+use function MongoDB\addLogger;
+use function MongoDB\removeLogger;
+
+/** @see https://jira.mongodb.org/browse/DRIVERS-2583 */
+class LogNonGenuineHostTest extends TestCase
+{
+    private LoggerInterface $logger;
+
+    public function setUp(): void
+    {
+        $this->logger = $this->createTestPsrLogger();
+
+        addLogger($this->logger);
+    }
+
+    public function tearDown(): void
+    {
+        removeLogger($this->logger);
+    }
+
+    /** @dataProvider provideCosmosUris */
+    public function testCosmosUriLogsInfoMessage(string $uri): void
+    {
+        $this->createClientAndIgnoreSrvLookupError($uri);
+
+        $expectedLog = [
+            'info',
+            'You appear to be connected to a CosmosDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/cosmosdb',
+            ['domain' => 'mongoc'],
+        ];
+
+        $this->assertContains($expectedLog, $this->logger->logs);
+    }
+
+    public function provideCosmosUris(): array
+    {
+        return [
+            ['mongodb://a.mongo.cosmos.azure.com:19555/'],
+            ['mongodb://a.MONGO.COSMOS.AZURE.COM:19555/'],
+            ['mongodb+srv://a.mongo.cosmos.azure.com/'],
+            ['mongodb+srv://A.MONGO.COSMOS.AZURE.COM/'],
+            // Mixing genuine and nongenuine hosts (unlikely in practice)
```

Comment:
Note: not using named data providers was intentional, since it would have involved a lot of redundancy. I only added comments to notable test cases from my original libmongoc tests.